### PR TITLE
step selector: deselect fob in step selector

### PR DESCRIPTION
**Pull request by japie1235813**
* Motivation for features / changes
The behavior of deselect fob still connects to linked time behavior (update "selectTimeEnableToggled", which is for linked time). We want to make the fob disappear when we deselect it in step selector single selection. This PR checks whether linked time is enabled and fires `stepSelectorEnableToggled` or `selectTimeEnableToggled` actions accordingly.

* Technical description of changes
1. fire `stepSelectorEnableToggled` action when linked time is not enabled. The "onSelectTimeToggle" is for linked time change selected time in ngrx state. When it's step selector, we do not have selected step in state (storing in each card level)
2. fix step selector checking bug: it's probably misunderstanding and misuse. `isScalarStepSelectorFeatureEnabled` is about wheteh the feature flag is turned on. `isScalarStepSelectorEnabled` is when users check/uncheck step selector.

* Demo

https://user-images.githubusercontent.com/1131010/176325540-24f330b2-37fa-4f57-9c55-4a5445940d14.mov

### CODE REVIEW

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts [hunk 1]
```diff
@@ -2616,6 +2620,17 @@ describe('scalar card', () => {
           end: null,
         });
       }));
+
+      it('toggles selected time when single fob is deselected in step selector', fakeAsync(() => {
```
Comment (bmd3k) [inline tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts:2624]: maybe: toggles step selector when single fob is deselected in step selector

File: tensorboard/webapp/metrics/views/right_pane/settings_view_component.ts [hunk 2]
```diff
@@ -65,6 +65,7 @@ export class SettingsViewComponent {
 
   @Input() isCardWidthSettingEnabled!: boolean;
   @Input() isLinkedTimeFeatureEnabled!: boolean;
+  @Input() isScalarStepSelectorFeatureEnabled!: boolean;
   @Input() isScalarStepSelectorEnabled!: boolean;
   @Input() selectTimeEnabled!: boolean;
```
Comment (bmd3k) [inline tensorboard/webapp/metrics/views/right_pane/settings_view_component.ts:70]: selectTimeEnabled => isLinkedTimeEnabled?

So you have isLinkedTimeFeatureEnabled and isLinkedTimeEnabled just like for ScalarStepSelector. This one might be worth doing in this PR but up to you.

**Review by bmd3k — APPROVED** [review]
I've called out a couple symbols that you might want to rename as part of the bigger rename PR we discussed offline; one symbol you could probably change in this PR; and then one suggestion to slightly rename a test.

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts (hunk 1 above)
Comment (japie1235813) [reply tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts]: removed "step selector" cause this is nested in the "step selector feature integration" test

File: tensorboard/webapp/metrics/views/right_pane/settings_view_component.ts (hunk 2 above)
Comment (japie1235813) [reply tensorboard/webapp/metrics/views/right_pane/settings_view_component.ts:70]: already started on the giant renaming PR before seeing this comment. Will put in following PR

**Pull request merged**